### PR TITLE
Add more validation to service struct

### PR DESCRIPTION
**Pull request by krhubert**
close https://github.com/mesg-foundation/engine/issues/1088

### CODE REVIEW

File: service/parameter.go [hunk 1]
```diff
@@ -22,5 +22,5 @@ type Parameter struct {
 	Repeated bool `hash:"name:6"`
 
 	// Definition of the structure of the object when the type is object
-	Object []*Parameter `hash:"name:7"`
+	Object []*Parameter `hash:"name:7" validate:"unique,dive,required"`
```
Comment (NicolasMahe) [inline service/parameter.go]: why this one is unique?
does it actually works?

File: service/service_validate.go [hunk 2]
```diff
@@ -82,6 +86,59 @@ func ValidateService(service *Service) error {
...
+	exist = make(map[string]bool)
+	for _, event := range service.Events {
+		if exist[event.Key] {
+			errs = append(errs, fmt.Errorf("events[%s] already exist", event.Key))
+		}
+		exist[event.Key] = true
+
+		existparam := make(map[string]bool)
+		for _, param := range event.Data {
+			if existparam[param.Key] {
```
Comment (NicolasMahe) [inline service/service_validate.go]: params.Object keys are not tested (task inputs, task outputs and event)
maybe we could use the validation lib with something like:
```
Object []*Parameter `hash:"name:7" validate:"dive,(key)unique,required"`
```

Comment (NicolasMahe) [reply service/service_validate.go]: otherwise a new custom validation like `PortMapping`

File: service/parameter.go (hunk 1 above)
Comment (krhubert) [reply service/parameter.go]: it won't be used here, but shouldn't exist

**Pull request merged**